### PR TITLE
Remove GenericOperation

### PR DESCRIPTION
**Pull request by Crazykev**
Don't need to go through the vm state machine.

/cc @laijs 

Signed-off-by: Crazykev <crazykev@zju.edu.cn>

### CODE REVIEW

File: hypervisor/vm.go [hunk 1]
```diff
@@ -358,7 +358,7 @@ func (vm *Vm) AddRoute() error {
 
 func (vm *Vm) AddNic(info *api.InterfaceDescription) error {
 	client := make(chan api.Result, 1)
-	vm.sendGenericOperation("CreateInterface", func(ctx *VmContext, result chan<- error) {
+	vm.handleGenericOperation("CreateInterface", func(ctx *VmContext, result chan<- error) {
 		go ctx.AddInterface(info, client)
```
Comment (laijs) [inline hypervisor/vm.go]: could you also remove vm.handleGenericOperation()?
we can use ctx.AddInterface() directly.

Comment (Crazykev) [inline hypervisor/vm.go]: @laijs fixed.

File: hypervisor/vm.go [hunk 2]
```diff
@@ -711,37 +715,29 @@ func (vm *Vm) GetIPAddrs() []string {
 	return ips
 }
 
-// sendGenericOperation queues a generic operation onto the VM's context if it
-// is in position to handle it.
-func (vm *Vm) sendGenericOperation(name string, op func(ctx *VmContext, result chan<- error), states ...string) <-chan error {
-	result := make(chan error, 1)
-
+func (vm *Vm) GenericOperation(name string, op func(ctx *VmContext, result chan<- error), states ...string) error {
```
Comment (laijs) [inline hypervisor/vm.go]: please keep sendGenericOperation() GenericOperation() unchanged until all its use-cases are converted.

Comment (Crazykev) [inline hypervisor/vm.go]: @laijs sure, now I just rename `sendGenericOperation()` to `handleGenericOperation()` (for what it really do), would this be fine?

**Comment by Crazykev** [comment]
@laijs This PR is ready now.

**Pull request merged**